consent/strategy_default.go

```diff
@@ -416,11 +416,12 @@ func (s *DefaultStrategy) requestConsent(w http.ResponseWriter, r *http.Request,
 	// accepted, as appropriate.
 	if ar.GetClient().IsPublic() {
 		// The OpenID Connect Test Tool fails if this returns `consent_required` when `prompt=none` is used.
-		// According to the quote above, it should be ok to allow https to skip consent.
+		// According to the quote above, it should be ok to allow https to skip consent. Additionally any localhost
+		// hostname is allowed to skip consent.
 		//
 		// This is tracked as issue: https://github.com/ory/hydra/issues/866
 		// This is also tracked as upstream issue: https://github.com/openid-certification/oidctest/issues/97
-		if ar.GetRedirectURI().Scheme != "https" {
+		if ar.GetRedirectURI().Scheme != "https" && !(fosite.IsLocalhost(ar.GetRedirectURI() && ar.GetRedirectURI().Scheme == "http")) {
```

Comment:
Seems like there's an issue with the parenthesis here:

```
!(fosite.IsLocalhost(ar.GetRedirectURI() && ar.GetRedirectURI().Scheme == "http"))
```

should probably be:

```
!(fosite.IsLocalhost(ar.GetRedirectURI()) && ar.GetRedirectURI().Scheme == "http"))
```